app/models/concerns/indexable.rb

```diff
@@ -279,19 +279,34 @@ def count
       Elasticsearch::Model.client.count(index: index_name)['count']
     end
 
+    def reindex(index: nil)
+      return nil unless index.present?
+      
+      self.__elasticsearch__.create_index! index: index unless self.__elasticsearch__.index_exists? index: index
+      index_name = self.index_name
+
+      client = Elasticsearch::Model.client
+      ### always take origing index even if it was a alias
+      index_name = client.indices.get_alias(name: index_name).first.first  if client.indices.exists_alias? name: index_name
+      client.reindex body: { source: { index: index_name }, dest: { index: index } }
+      puts "Index #{index} exists? #{self.__elasticsearch__.index_exists? index: index}"
+    end
+
     def create_alias(index: nil)
       return nil unless index.present?
 
-      client     = self.gateway.client
+      client     = Elasticsearch::Model.client
       index_name = self.index_name
 
       client.indices.put_alias index: index, name: index_name
+      puts client.indices.get_alias name: index_name
```

Comment:
Why is there a `puts`? Do you want to put this in the log? If so, it may need more context.

Reply:
Well yeah, I wanted to show some information about the effects of the changes.  I can add more  context on it.

Reply:
I would use `puts` only in rake tasks or temporarily. Otherwise use the logger.

Reply:
this has been changed